lib/src/mt_ptp.c

```diff
@@ -103,11 +118,15 @@ static inline void ptp_set_master_addr(struct mt_ptp_impl* ptp,
 }
 
 static void ptp_adjust_delta(struct mt_ptp_impl* ptp, int64_t delta) {
+  uint64_t ts_s = ptp_get_raw_time(ptp);
+  uint64_t ts_m = ts_s + delta;
+  ptp->coefficient = (double)(ts_m - ptp->last_sync_ts) / (ts_s - ptp->last_sync_ts);
   ptp_timesync_lock(ptp);
   rte_eth_timesync_adjust_time(ptp->port_id, delta);
   ptp_timesync_unlock(ptp);
-  dbg("%s(%d), delta %" PRId64 ", ptp %" PRIu64 "\n", __func__, ptp->port, delta,
-      ptp_get_raw_time(ptp));
+  ptp->last_sync_ts = ptp_get_raw_time(ptp);
```

Comment:
Directly use "ptp->last_sync_ts = ts_s"?

Reply:
ts_s is before sync and not accurate, maybe ts_m or ptp_correct_ts(ts_s)?